### PR TITLE
LPD-34378 - Add ERC to email aderesses

### PR DESCRIPTION
**Pull request by matyaswollner**
https://liferay.atlassian.net/browse/LPD-34378

### CODE REVIEW

File: portal-impl/src/com/liferay/portal/service/impl/EmailAddressLocalServiceImpl.java [hunk 1]
```diff
@@ -31,13 +31,18 @@ public class EmailAddressLocalServiceImpl
...
+			ServiceContext serviceContext)
 		throws PortalException {
 
 		User user = _userPersistence.findByPrimaryKey(userId);
 		long classNameId = _classNameLocalService.getClassNameId(className);
 
+		if (Validator.isBlank(externalReferenceCode)) {
+			externalReferenceCode = null;
+		}
+
```
Comment (smottal) [inline portal-impl/src/com/liferay/portal/service/impl/EmailAddressLocalServiceImpl.java:45]: Please move this at the beginning of the method

File: portal-impl/src/com/liferay/portlet/usersadmin/util/UsersAdminUtil.java [hunk 2]
```diff
@@ -1477,14 +1477,14 @@ public static void updateEmailAddresses(
 
 			if (emailAddressId <= 0) {
 				emailAddress = EmailAddressServiceUtil.addEmailAddress(
-					className, classPK, address, listTypeId, primary,
+					null, className, classPK, address, listTypeId, primary,
```
Comment (smottal) [inline portal-impl/src/com/liferay/portlet/usersadmin/util/UsersAdminUtil.java:1480]: probably we have to get the erc from the emailAddress no?

File: portal-impl/src/com/liferay/portlet/usersadmin/util/UsersAdminUtil.java [hunk 3]
```diff
@@ -1477,14 +1477,14 @@ public static void updateEmailAddresses(
...
 					new ServiceContext());
 
 				emailAddressId = emailAddress.getEmailAddressId();
 			}
 			else {
 				EmailAddressServiceUtil.updateEmailAddress(
-					emailAddressId, address, listTypeId, primary);
+					null, emailAddressId, address, listTypeId, primary);
```
Comment (smottal) [inline portal-impl/src/com/liferay/portlet/usersadmin/util/UsersAdminUtil.java:1487]: same here

File: portal-impl/src/com/liferay/portal/upgrade/v7_4_x/UpgradeEmailAddressExternalReferenceCode.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+package com.liferay.portal.upgrade.v7_4_x;
+
+import com.liferay.portal.kernel.upgrade.BaseExternalReferenceCodeUpgradeProcess;
+
+/**
+ * @author Matyas Wollner
+ */
+
+public class UpgradeEmailAddressExternalReferenceCode extends BaseExternalReferenceCodeUpgradeProcess {
```
Comment (smottal) [inline portal-impl/src/com/liferay/portal/upgrade/v7_4_x/UpgradeEmailAddressExternalReferenceCode.java:14]: Wrong name: let's call it EmailAddressExternalReferenceCodeUpgradeProcess.java

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:relevant

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:sf

**Pull request closed without merging**